design/legacy-design/device-service/discovery.md

```diff
@@ -0,0 +1,79 @@
+## Dynamic Device Discovery
+
+### Overview
+
+Some device protocols allow for devices to be discovered automatically. A Device
+Service may include a capability for discovering devices and creating the
+corresponding Device objects within EdgeX. A framework for doing so will be
+implemented in the Device Service SDKs.
+
+The discovery process will operate as follows:
+
+* Discovery is triggered either on an internal timer or by a call to a REST endpoint
+* The SDK will call a function provided by the DS implementation to request a
+device scan
+* The implementation calls back to the SDK with details of devices which it has found
+* The SDK filters these devices against a set of acceptance criteria
+* The SDK adds accepted devices in core-metadata. These are now available in
+the EdgeX system
+
+### Triggering Discovery
+
+A boolean configuration value `Device/Discovery` defaults to false. If this
+value is set true, and the DS implementation supports discovery, discovery is
+enabled.
+
+The SDK will respond to POST requests on the the /discovery endpoint. This call
+will return one of the following codes:
+
+* 200: discovery has been triggered or is already running
+* 423: the service is locked (admin state) or disabled (operating state)
+* 500: unknown or unanticipated issues exist
+* 501: discovery is not supported in this implementation
+* 503: discovery is disabled by configuration
+
+In the case where discovery is triggered, the discovery process will run in a
+new thread or goroutine, so that the REST call may return immediately.
+
+An integer configuration value `Device/DiscoveryInterval` defaults to zero. If
+this value is set to a positive value, and discovery is enabled, the discovery
+process will be triggered at the specified interval (in seconds).
```

Comment:
Recommend `Discovery` be its own section in the configuration. Example:

```
Device/Discovery/Enabled = true
Device/Discovery/Interval = 3 (seconds)
```

Reply:
Agreed